### ISSUE TITLE
split api and implementatin

### ISSUE THREAD

**Issue by judovana** [terminus-brut/jc#6]
main ide aof the split is, that api can be jdk8 only. Current implementation is not.(and that is probably good:)

**Comment by judovana** [comment]
Probably ideal would be to have it as two maven modules.
If you wish, I can do that. And will rework the repo a bit so the parent pom is not needed and that assembly and build  works much more flawless. Can I?

**Comment by judovana** [comment]
Yes. This is about splitting the repo to two modules - each with its own jar and pom. And probably with top level pom 
The reason is, that the api is stable, but the implementation is heavily jdk dependent. Had you tried current one with jdk17?

Eg right now, api can be compiled by jdk8,  where your implementation needs jdk11.  As it is now, you force users of ap to be on jdk11, ehre they still could be on 8, and just drop some functionality when not being on 11.

**Comment by mkoncek** [comment]
See https://github.com/mkoncek/classpathless-compiler/tree/e0adb74362a64658345906a86ba617101f846147.
With the split in api, it is now possible to separate versions of api and impl.
